### ISSUE TITLE
Tusimple Dataset Training Problem

### ISSUE THREAD

**Issue by yxxxxxxxx** [cardwing/Codes-for-Lane-Detection#31]
Hello! I am trying your repo on Tusimple Dataset. I have met a problem which looks like as following:
![image](https://user-images.githubusercontent.com/26196715/50590904-cfda9400-0ec7-11e9-8f6f-7a93447b5a6b.png)
It seems that all the output becomes all zero. I wonder if you have met this problem and how can I solve this problem! I just use your list and adjust the input size to 256 x 512. My tf verson is 1.10.0.  Thank you very much!

**Comment by cardwing** [comment]
@yxxxxxxxx, it is common that the accuracy will become 0 in the first few episodes. Just train more episodes and see if the output is always zero. Besides, you can also check if the input labels are correct (it should consist of 0,1,2,3,4).

**Comment by yxxxxxxxx** [comment]
Thank you for your reply! I've tried many times and trained more than  90000 iters. It is still zero and the loss haven't changed since a few iters at the beginning. I also check the label image which indeed consist of 0,1,2,3,4 at different pixels.  So it really troubles me since I use the same global_config.py except the IMG_HEIGHT and IMG_WIDTH set to 256 and 512 respectively.

**Comment by cardwing** [comment]
@yxxxxxxxx, I have also met this problem after retraining the model using the code. Thanks for pointing it out. You can use the old code (before Dec 19, 2018) which should be ok. I will check the problem ASAP.

**Comment by cardwing** [comment]
@yxxxxxxxx, the following version of the code should be ok.

![code_v](https://user-images.githubusercontent.com/23656119/50596543-9e6dc280-0edf-11e9-8f1e-587f18a0894a.png)

**Comment by cardwing** [comment]
@yujincheng08, please help check this problem too. Thanks a lot.

**Comment by yxxxxxxxx** [comment]
> @yxxxxxxxx, the following version of the code should be ok.
> 
> ![code_v](https://user-images.githubusercontent.com/23656119/50596543-9e6dc280-0edf-11e9-8f1e-587f18a0894a.png)

Thank you very much! I will try it and continue to figure out this problem!

**Comment by yujincheng08** [comment]
@yxxxxxxxx Hi! Could you please try my code? My fork seems to work pretty good and I already trained my own model using my code. I think there might be some inconsistency between my code and the original one.  
https://github.com/yujincheng08/Codes-for-Lane-Detection

**Comment by yxxxxxxxx** [comment]
@yujincheng08 @cardwing Thank you for your help! It seems to work when I use @yujincheng08 fork.

**Comment by yujincheng08** [comment]
@yxxxxxxxx Thanks for your feedback.
@cardwing I think the easiest way to fix this issue is to merge my fork. How do you think?

**Comment by cardwing** [comment]
@yujincheng08, it is indeed a convenient solution. I will have a quick test on your code tomorrow and merge your fork if it works well.

**Comment by yxxxxxxxx** [comment]
> @yxxxxxxxx, what is the final performance (accuracy) of SCNN-Tensorflow on TuSimple testing set? If you can provide me with the result, it can save me some time.

I'm sorry for late reply! I haven't test the final performance. I notice that the Tusimple Dataset has up to 6 lanes in an image but the list you offered has just 4 lanes. In #4, the pred_json.py evaluate each image by 6 lanes. If so, should I just label this dataset by myself of you can provide a more accurate label list?

**Comment by cardwing** [comment]
@yxxxxxxxx, the following package contains the 6-lane-annotation lists, please check.

[new_list_tusimple.zip](https://github.com/cardwing/Codes-for-Lane-Detection/files/2731505/new_list_tusimple.zip)

**Comment by yxxxxxxxx** [comment]
> @yxxxxxxxx, the following package contains the 6-lane-annotation lists, please check.
> 
> [new_list_tusimple.zip](https://github.com/cardwing/Codes-for-Lane-Detection/files/2731505/new_list_tusimple.zip)

Ok, thank you!

**Comment by MaybeShewill-CV** [comment]
@yxxxxxxxx Looking forward to your test result. Thanks for sharing:)
